recipes/dcv_install.rb

```diff
@@ -70,7 +70,7 @@ def disable_lock_screen
   not_if { ::File.exist?("#{node['cfncluster']['scripts_dir']}/pcluster_dcv_connect.sh") }
```

Comment:
with the.new logic I think this cookbook_file resource is skipped, but afaik it was intended to be there even for not supported OS, like comment says "# Install pcluster_dcv_connect.sh script in all the OSes to use it for error handling"

Reply:
Wouldn't it only be skipped if it already exists?

The guards at the top of the file only prevent the recipe from running again if the AMI's bootstrapped or if /etc/dcv/dcv.conf exists (which I believe implies this has already run anyway).

The recipe itself is source in two recipes:
* from dcv_config, 
* from default

The include in default doesn't have any guards around it, so that shouldn't be a problem. I believe this is the entrypoint to the cookbook for both the AMI build process and for runtime AMI baking.

The only process where this would be an issue would be the kitchen tests. Does it matter that it's not being included there? If DCV isn't supported on the platform it's going to skip all of those tests anyways.

Reply:
ok you are right. Probably the check on `/etc/dcv/dcv.conf"` could be even removed since there is the one on the `ami_bootstrapped `.

I didn't get why this would be an issue in the kitchen tests

Reply:
It would be an issue during kitchen tests because the include path for that scenario is through dcv_config, which would prevent this recipe from running because it checks conditions.

I don't think that would be an issue since I assume none of the kitchen tests are exercising that script.